server/call/operator.py

```diff
@@ -0,0 +1,63 @@
+"""Module which keeps track of all ongoing sessions and provides
+querying functionality to HTTP requesters."""
+import threading
+
+from daily import Daily
+from server.config import Config
+from server.call.session import Session
+
+
+class Operator():
+    _config: Config
+    _sessions: list[Session] = []
+    _is_shutting_down: bool
+
+    def __init__(self, config: Config):
+        self._config = config
+        self._is_shutting_down = False
+        Daily.init()
+
+    def create_session(self, room_duration_mins: int = None) -> str:
+        """Creates a session, which includes creating a Daily room."""
+        session = Session(self._config, self.clear_session, room_duration_mins)
+        self._sessions.append(session)
+        print("CREATED SESSION. ALL THREADS:", threading.active_count())
+        for thread in threading.enumerate():
+            print(thread.name, thread.is_alive())
+
+        return session.room_url
+
+    def query_assistant(self, room_url: str, custom_query=None) -> str:
+        """Queries the assistant for the provided room URL."""
+        for s in self._sessions:
+            if s.room_url == room_url:
+                return s.query_assistant(custom_query=custom_query)
+        raise Exception(
+            f"Requested room URL {room_url} not found in active sessions")
+
+    def clear_session(self, id: str, room_url: str):
+        """Removes the session from the list of active sessions."""
+        for s in self._sessions:
+            if s.id == id and s.room_url == room_url:
+                self._sessions.remove(s)
```

Comment:
There's a circular dependency here. The `Session` is telling the `Operator` to remove itself from the list of sessions. This is happening while the `Session` is executing `on_left_meeting`. In practice there will probably be no issues because the garbage collector won't get rid of the Session right away. But still it might be a good idea to do this differently. One possible solution would be to have a housekeeping thread running in the Operator that periodically checks for finished sessions. A finished session flag would be set inside `on_left_meeting`.

Reply:
Thanks, I've gone with having the operator run a cleanup check every 5 seconds

Reply:
Sounds good! 🙌